conjure-runtime/src/service/raw/js.rs

```diff
@@ -0,0 +1,226 @@
+// Copyright 2025 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+use crate::service::raw::RawRequestBody;
+use crate::service::Service;
+use crate::{builder, Builder};
+use bytes::Bytes;
+use conjure_error::Error;
+use http::{header, HeaderName, HeaderValue, Request, Response, StatusCode};
+use http_body::{Body, Frame};
+use http_body_util::BodyExt;
+use js_sys::{Array, JsString, Promise, Uint8Array};
+use std::cell::RefCell;
+use std::convert::TryFrom;
+use std::future::Future;
+use std::pin::Pin;
+use std::rc::Rc;
+use std::task::{ready, Context, Poll};
+use std::{error, fmt};
+use wasm_bindgen::prelude::{wasm_bindgen, Closure, JsCast, JsValue};
+use wasm_bindgen_futures::JsFuture;
+use web_sys::{
+    AbortController, Headers, ReadableStream, ReadableStreamDefaultController,
+    ReadableStreamDefaultReader, ReadableStreamReadResult, RequestInit, UnderlyingSource,
+};
+
+static FETCH_USER_AGENT: HeaderName = HeaderName::from_static("fetch-user-agent");
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_name = fetch)]
+    fn fetch_with_request(input: &web_sys::Request) -> Promise;
+}
+
+pub struct RawClient(());
+
+impl RawClient {
+    pub fn new(_: &Builder<builder::Complete>) -> Result<Self, Error> {
+        Ok(RawClient(()))
+    }
+}
+
+impl Service<Request<RawRequestBody>> for RawClient {
+    type Response = Response<RawResponseBody>;
+    type Error = Box<dyn error::Error + Sync + Send>;
+
+    // The fetch API promises to call these futures sequentially
+    #[allow(clippy::await_holding_refcell_ref)]
+    async fn call(&self, req: Request<RawRequestBody>) -> Result<Self::Response, Self::Error> {
+        let (parts, body) = req.into_parts();
+
+        let init = RequestInit::new();
+        init.set_method(parts.method.as_str());
+
+        let headers = Headers::new().map_err(JsError::new)?;
+        for (mut name, value) in &parts.headers {
+            if name == header::USER_AGENT {
+                name = &FETCH_USER_AGENT;
+            }
```

Comment:
Chrome doesn't let you override the user-agent header, so we use the same special `fetch-user-agent` header that the typescript client does and Witchcraft servers know to look for.